### PR TITLE
Exception Page showing only method names in the callstack

### LINKED ISSUE #304: THIS ISSUE TRACKER IS CLOSED - use the Home repo issue tracker
Hello, this issue tracker is no longer used. Please log new issues here: https://github.com/aspnet/Home/issues

For information about this change, see https://github.com/aspnet/Announcements/issues/283

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #304

### CODE REVIEW

File: test/Microsoft.Extensions.Internal.Test/StackTraceTest.cs [hunk 1]
```diff
@@ -14,17 +19,27 @@ public class StackTraceTest
...
         };
 
         [Theory]
         [MemberData(nameof(CanGetStackTraceData))]
-        public void GetFrames_CanGetStackTrace(Action action, string expectedMethodName)
+        public void GetFrames_CanGetStackTrace(Action action, string expectedDisplay)
         {
             try
             {
+                //
```
Comment (halter73) [inline test/Microsoft.Extensions.Internal.Test/StackTraceTest.cs]: Nit: did you mean to add a comment here?

File: src/Microsoft.Extensions.StackTrace.Sources/StackTraceHelper.cs [hunk 2]
```diff
@@ -58,5 +61,75 @@ public static IList<StackFrameInfo> GetFrames(Exception exception)
...
+                return null;
+            }
+
+            var methodDisplayInfo = new MethodDisplayInfo();
+
+            // Type name
+            var type = method.DeclaringType;
+            if (type != null)
+            {
+                var typeName = TypeNameHelper.GetTypeDisplayName(type, fullName: false);
```
Comment (halter73) [inline src/Microsoft.Extensions.StackTrace.Sources/StackTraceHelper.cs]: Can `fullName: false` mess up the display name of nested classes?

File: test/Microsoft.Extensions.Internal.Test/StackTraceTest.cs [hunk 3]
```diff
@@ -57,13 +75,69 @@ public void GetFrames_DoesNotFailForDynamicallyGeneratedAssemblies()
...
                 // Assert
                 Assert.Null(frames.First().FilePath);
-                Assert.Equal(nameof(GetFrames_DoesNotFailForDynamicallyGeneratedAssemblies), frames.Last().Method);
+                Assert.Equal(
+                    $"{typeof(StackTraceTest).GetTypeInfo().FullName}.{nameof(GetFrames_DoesNotFailForDynamicallyGeneratedAssemblies)}()",
+                    frames.Last().MethodDisplayInfo.ToString());
             }
         }
 
+        public static TheoryData GetMethodDisplayString_ReturnsTypeNameQualifiedMethodsData
```
Comment (halter73) [inline test/Microsoft.Extensions.Internal.Test/StackTraceTest.cs]: Can we add test cases for nested classes defining both the method and argument types?

Comment (pranavkm) [inline test/Microsoft.Extensions.Internal.Test/StackTraceTest.cs]: Done. It showed an issue with the code we have with Hosting today where the stack trace line we generate for nested types don't include the containing object.

**Comment by pranavkm** [comment]
:up: :date:

**Comment by halter73** [comment]
:shipit:

**Pull request closed without merging**